IdentityCore/src/webview/systemWebview/ios/MSIDAuthenticationSession.m

```diff
@@ -88,29 +117,36 @@ - (void)startWithCompletionHandler:(MSIDWebUICompletionHandler)completionHandler
         
         _completionHandler = [completionHandler copy];
         
-        _authSession = [[SFAuthenticationSession alloc] initWithURL:_startURL
-                                                  callbackURLScheme:_callbackURLScheme
-                                                  completionHandler:^(NSURL * _Nullable callbackURL, NSError * _Nullable error)
-                                            {
-                                                if (error)
-                                                {
-                                                    if (error.code == SFAuthenticationErrorCanceledLogin)
-                                                    {
-                                                        error = MSIDCreateError(MSIDErrorDomain, MSIDErrorUserCancel, @"User cancelled the authorization session.", nil, nil, nil, _context.correlationId, nil);
-                                                        [_telemetryEvent setIsCancelled:YES];
-                                                    }
-                                                }
-
-                                                [[MSIDTelemetry sharedInstance] stopEvent:_telemetryRequestId event:_telemetryEvent];
-                                                
-                                                [self notifyEndWebAuthWithURL:callbackURL error:error];
-                                                _completionHandler(callbackURL, error);
-                                            }];
+        void (^authCompletion)(NSURL *, NSError *) = ^void(NSURL *callbackURL, NSError *authError)
+        {
+            error = [self errorFromAuthResponse:authError];
+            if (error && error.code == MSIDErrorUserCancel)
```

Comment:
would there be any other error code every returned? I don't like relying on MSIDErrorUserCancel here...

Reply:
If there are any other error code returned, it would be handled without telemetryEvent setIsCancelled at line 130.
This one is to basically set telemetryEvent and that'd be it.

Reply:
Maybe we can set telemetry event when creating the error?

Reply:
Changes so that helper methods returns a Bool on whether or not it is user cancelled.